lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -70,13 +70,35 @@ def last_inserted_id(result)
 
       include DatabaseStatements
 
+      SSL_MODES = {
+        SSL_MODE_DISABLED: Trilogy::SSL_DISABLED,
+        SSL_MODE_PREFERRED: Trilogy::SSL_PREFERRED_NOVERIFY,
+        SSL_MODE_REQUIRED: Trilogy::SSL_REQUIRED_NOVERIFY,
+        SSL_MODE_VERIFY_CA: Trilogy::SSL_VERIFY_CA,
+        SSL_MODE_VERIFY_IDENTITY: Trilogy::SSL_VERIFY_IDENTITY
+      }.freeze
+
       class << self
         def new_client(config)
+          config[:ssl_mode] = parse_ssl_mode(config[:ssl_mode]) if config[:ssl_mode]
           ::Trilogy.new(config)
         rescue Trilogy::DatabaseError => error
           raise translate_connect_error(config, error)
         end
 
+        def parse_ssl_mode(mode)
+          return unless mode
```

Comment:
Is this redundant with `if config[:ssl_mode]` above?

Reply:
I thought we could use an extra check since this method could technically be called anywhere but maybe a single check is good. I'll update it 😃